Src/AutoFixture/MailAddressGenerator.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Net.Mail;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Creates new <see cref="MailAddress"/> instances.
+    /// </summary>
+    public class MailAddressGenerator : ISpecimenBuilder
+    {
+        private readonly string[] fictitiousDomains = 
+        { 
+            "example.com", 
+            "example.net", 
+            "example.org" 
+        };
+
+        /// <summary>
+        /// Creates a new anonymous MailAddress.
+        /// </summary>
+        /// <param name="request">The request that describes what to create.</param>
+        /// <param name="context">A context that can be used to create other specimens.</param>
+        /// <returns>
+        /// The requested specimen if possible; otherwise a <see cref="NoSpecimen"/> instance.
+        /// </returns>
+        /// <remarks>>
+        /// The generated MailAddress will have one of the reserved domains,
+        /// so as to avoid any possibility of tests bothering real email addresses
+        /// </remarks>
+        public object Create(object request, ISpecimenContext context)
+        {
+            if (!typeof(MailAddress).Equals(request))
+            {
+                return new NoSpecimen(request);
+            }
+
+            var name = Guid.NewGuid().ToString();

```

Comment:
Yes, similar to [this](https://github.com/AutoFixture/AutoFixture/pull/340#discussion_r23566830) comment.